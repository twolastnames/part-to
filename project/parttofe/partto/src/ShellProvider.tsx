import React, { MutableRefObject, useRef, useState } from "react";

import { MantineProvider, createTheme } from "@mantine/core";
import { PropsWithChildren } from "react";
import "@mantine/core/styles.layer.css";
import "./App.scss";

const themes = {
  default: createTheme({
    primaryColor: "red",
  }),
  dark: createTheme({
    primaryColor: "red",
  }),
};

let themeChangeRef: MutableRefObject<(theme: Theme) => void>;

export enum Theme {
  Default = "default",
  Dark = "dark",
}

export function changeTheme(theme: Theme) {
  const themeNode = document.getElementById("theme");
  if (!themeNode || !themeNode.className) {
    console.error("developers should have created an theme ID in the project");
    return;
  }
  themeChangeRef?.current && themeChangeRef.current(theme);
  themeNode.className = `${theme}Theme`;
}

export const ShellProvider = ({ children }: PropsWithChildren) => {
  const [theme, setTheme] = useState<Theme>(Theme.Default);
  themeChangeRef = useRef(setTheme);
  return (
    <div id="theme" className="defaultTheme">
      <MantineProvider theme={themes[theme]}>{children}</MantineProvider>
    </div>
  );
};
