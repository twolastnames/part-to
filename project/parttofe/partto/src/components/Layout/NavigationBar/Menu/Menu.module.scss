@import "../../../../App.scss";

.navLink:hover {
  background-color: var(--background-color);
}
