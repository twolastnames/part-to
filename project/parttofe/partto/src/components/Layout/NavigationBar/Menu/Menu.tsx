import React from "react";

import classes from "./Menu.module.scss";
import { NavLink } from "@mantine/core";
import { Book, ChefHat, Icon, Info, Oven, Settings } from "../../../Icon/Icon";
import { ThemeSwitch } from "../ThemeSwitch/ThemeSwitch";

export function Menu() {
  return (
    <>
      <nav data-testid="Menu" className={classes.nav}>
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Enter Recipe</div>}
          leftSection={<Icon definition={ChefHat} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Start Meal</div>}
          leftSection={<Icon definition={Book} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Continue Cooking</div>}
          leftSection={<Icon definition={Oven} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>Settings</div>}
          leftSection={<Icon definition={Settings} />}
        />
        <NavLink
          key={Math.random()}
          className={classes.navLink}
          label={<div className={classes.text}>About</div>}
          leftSection={<Icon definition={Info} />}
        />
      </nav>
      <ThemeSwitch key={Math.random()} />
    </>
  );
}
