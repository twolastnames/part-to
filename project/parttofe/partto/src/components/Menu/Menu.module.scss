@import "../../App.scss";

.items {
  animation: fadein var(--animation-tempo);
}
