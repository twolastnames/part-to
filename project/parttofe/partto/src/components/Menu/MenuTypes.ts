export interface MenuProps {
  runState: string;
}
