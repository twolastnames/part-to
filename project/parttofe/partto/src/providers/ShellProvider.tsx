import React, { MutableRefObject, useRef, useState } from "react";

import { MantineProvider, createTheme } from "@mantine/core";
import { PropsWithChildren } from "react";
import "@mantine/core/styles.layer.css";
import "../App.scss";
import {
  LeftDynamicItemSetPairProvider,
  RightDynamicItemSetPairProvider,
} from "./DynamicItemSetPair";
import { TimerProvider } from "./Timer";

export enum Theme {
  Default = "default",
  Dark = "dark",
}

const themeStorageKey = "theme";

export function getTheme(): Theme {
  return (localStorage.getItem(themeStorageKey) || Theme.Default) as Theme;
}

const themes = {
  default: createTheme({
    primaryColor: "red",
  }),
  dark: createTheme({
    primaryColor: "red",
  }),
};

let themeChangeRef: MutableRefObject<(theme: Theme) => void>;

function setThemeClass() {
  const themeNode = document.getElementById("theme");
  if (!themeNode || !themeNode.className) {
    console.error("developers should have created an theme ID in the project");
    return;
  }
  themeNode.className = `${getTheme()}Theme`;
}

export function changeTheme(theme: Theme) {
  localStorage.setItem(themeStorageKey, theme.toString());
  themeChangeRef?.current && themeChangeRef.current(theme);
}

export const ShellProvider = ({ children }: PropsWithChildren) => {
  const [theme, setTheme] = useState<Theme>(getTheme());
  themeChangeRef = useRef(setTheme);
  setTimeout(() => {
    setThemeClass();
  }, 0);
  return (
    <div id="theme" className="defaultTheme">
      <LeftDynamicItemSetPairProvider>
        <RightDynamicItemSetPairProvider>
          <TimerProvider>
            <MantineProvider theme={themes[theme]}>{children}</MantineProvider>
          </TimerProvider>
        </RightDynamicItemSetPairProvider>
      </LeftDynamicItemSetPairProvider>
    </div>
  );
};
