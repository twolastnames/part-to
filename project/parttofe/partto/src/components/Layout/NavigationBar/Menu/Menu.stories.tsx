import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { Menu } from "./Menu";
import { ShellProvider } from "../../../../ShellProvider";

const meta: Meta<typeof Menu> = {
  component: Menu,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof Menu>;

export const Simple: Story = {
  args: {},
};
