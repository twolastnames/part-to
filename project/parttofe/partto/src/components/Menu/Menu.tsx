import React from "react";

import classes from "./Menu.module.scss";
import { MenuProps } from "./MenuTypes";
import { useRunGet } from "../../api/runget";
import { Spinner } from "../Spinner/Spinner";
import { Accordion } from "../Accordion/Accordion";
import { useParttoGet } from "../../api/parttoget";

const Row = ({ partTo }: { partTo: string }) => {
  const response = useParttoGet({ partTo });
  return (
    <Spinner responses={[response]}>
      <li>{response.data?.name}</li>
    </Spinner>
  );
};

export function Menu({ runState }: MenuProps) {
  const response = useRunGet({ runState });

  return (
    <Spinner responses={[response]}>
      <Accordion summary="Menu">
        <ul data-testid="Menu" className={classes.items}>
          {response.data?.activePartTos?.map((partTo) => (
            <Row partTo={partTo} />
          ))}
        </ul>
      </Accordion>
    </Spinner>
  );
}
