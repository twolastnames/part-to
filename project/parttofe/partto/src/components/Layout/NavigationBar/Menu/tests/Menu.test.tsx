import React from "react";
import { expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { ShellProvider } from "../../../../../ShellProvider";
import { Menu } from "../Menu";

test("snapshot", () => {
  render(
    <ShellProvider>
      <Menu />
    </ShellProvider>,
  );
  const component = screen.getByTestId("Menu");
  expect(component).toMatchSnapshot();
});
