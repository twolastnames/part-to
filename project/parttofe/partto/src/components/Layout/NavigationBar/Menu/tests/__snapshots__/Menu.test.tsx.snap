// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<nav
  class="nav"
  data-testid="Menu"
>
  <a
    class="mantine-focus-auto navLink m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
  >
    <span
      class="m_690090b5 mantine-NavLink-section"
      data-position="left"
    >
      <svg
        class="tabler-icon tabler-icon-chef-hat "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3c1.918 0 3.52 1.35 3.91 3.151a4 4 0 0 1 2.09 7.723l0 7.126h-12v-7.126a4 4 0 1 1 2.092 -7.723a4 4 0 0 1 3.908 -3.151z"
        />
        <path
          d="M6.161 17.009l11.839 -.009"
        />
      </svg>
    </span>
    <div
      class="m_f07af9d2 mantine-NavLink-body"
    >
      <span
        class="m_1f6ac4c4 mantine-NavLink-label"
      >
        <div
          class="text"
        >
          Enter Recipe
        </div>
      </span>
      <span
        class="m_57492dcc mantine-NavLink-description"
      />
    </div>
  </a>
  <div
    aria-hidden="true"
    class="mantine-NavLink-collapse"
    style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
  >
    <div
      class="m_e17b862f mantine-NavLink-children"
    />
  </div>
  <a
    class="mantine-focus-auto navLink m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
  >
    <span
      class="m_690090b5 mantine-NavLink-section"
      data-position="left"
    >
      <svg
        class="tabler-icon tabler-icon-book "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 19a9 9 0 0 1 9 0a9 9 0 0 1 9 0"
        />
        <path
          d="M3 6a9 9 0 0 1 9 0a9 9 0 0 1 9 0"
        />
        <path
          d="M3 6l0 13"
        />
        <path
          d="M12 6l0 13"
        />
        <path
          d="M21 6l0 13"
        />
      </svg>
    </span>
    <div
      class="m_f07af9d2 mantine-NavLink-body"
    >
      <span
        class="m_1f6ac4c4 mantine-NavLink-label"
      >
        <div
          class="text"
        >
          Start Meal
        </div>
      </span>
      <span
        class="m_57492dcc mantine-NavLink-description"
      />
    </div>
  </a>
  <div
    aria-hidden="true"
    class="mantine-NavLink-collapse"
    style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
  >
    <div
      class="m_e17b862f mantine-NavLink-children"
    />
  </div>
  <a
    class="mantine-focus-auto navLink m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
  >
    <span
      class="m_690090b5 mantine-NavLink-section"
      data-position="left"
    >
      <svg
        class="tabler-icon tabler-icon-cooker "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 7h.01"
        />
        <path
          d="M15 7h.01"
        />
        <path
          d="M9 7h.01"
        />
        <path
          d="M5 3m0 2a2 2 0 0 1 2 -2h10a2 2 0 0 1 2 2v14a2 2 0 0 1 -2 2h-10a2 2 0 0 1 -2 -2z"
        />
        <path
          d="M9 15h6"
        />
        <path
          d="M5 11h14"
        />
      </svg>
    </span>
    <div
      class="m_f07af9d2 mantine-NavLink-body"
    >
      <span
        class="m_1f6ac4c4 mantine-NavLink-label"
      >
        <div
          class="text"
        >
          Continue Cooking
        </div>
      </span>
      <span
        class="m_57492dcc mantine-NavLink-description"
      />
    </div>
  </a>
  <div
    aria-hidden="true"
    class="mantine-NavLink-collapse"
    style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
  >
    <div
      class="m_e17b862f mantine-NavLink-children"
    />
  </div>
  <a
    class="mantine-focus-auto navLink m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
  >
    <span
      class="m_690090b5 mantine-NavLink-section"
      data-position="left"
    >
      <svg
        class="tabler-icon tabler-icon-settings "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.325 4.317c.426 -1.756 2.924 -1.756 3.35 0a1.724 1.724 0 0 0 2.573 1.066c1.543 -.94 3.31 .826 2.37 2.37a1.724 1.724 0 0 0 1.065 2.572c1.756 .426 1.756 2.924 0 3.35a1.724 1.724 0 0 0 -1.066 2.573c.94 1.543 -.826 3.31 -2.37 2.37a1.724 1.724 0 0 0 -2.572 1.065c-.426 1.756 -2.924 1.756 -3.35 0a1.724 1.724 0 0 0 -2.573 -1.066c-1.543 .94 -3.31 -.826 -2.37 -2.37a1.724 1.724 0 0 0 -1.065 -2.572c-1.756 -.426 -1.756 -2.924 0 -3.35a1.724 1.724 0 0 0 1.066 -2.573c-.94 -1.543 .826 -3.31 2.37 -2.37c1 .608 2.296 .07 2.572 -1.065z"
        />
        <path
          d="M9 12a3 3 0 1 0 6 0a3 3 0 0 0 -6 0"
        />
      </svg>
    </span>
    <div
      class="m_f07af9d2 mantine-NavLink-body"
    >
      <span
        class="m_1f6ac4c4 mantine-NavLink-label"
      >
        <div
          class="text"
        >
          Settings
        </div>
      </span>
      <span
        class="m_57492dcc mantine-NavLink-description"
      />
    </div>
  </a>
  <div
    aria-hidden="true"
    class="mantine-NavLink-collapse"
    style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
  >
    <div
      class="m_e17b862f mantine-NavLink-children"
    />
  </div>
  <a
    class="mantine-focus-auto navLink m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
  >
    <span
      class="m_690090b5 mantine-NavLink-section"
      data-position="left"
    >
      <svg
        class="tabler-icon tabler-icon-info-square-rounded "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 9h.01"
        />
        <path
          d="M11 12h1v4h1"
        />
        <path
          d="M12 3c7.2 0 9 1.8 9 9s-1.8 9 -9 9s-9 -1.8 -9 -9s1.8 -9 9 -9z"
        />
      </svg>
    </span>
    <div
      class="m_f07af9d2 mantine-NavLink-body"
    >
      <span
        class="m_1f6ac4c4 mantine-NavLink-label"
      >
        <div
          class="text"
        >
          About
        </div>
      </span>
      <span
        class="m_57492dcc mantine-NavLink-description"
      />
    </div>
  </a>
  <div
    aria-hidden="true"
    class="mantine-NavLink-collapse"
    style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
  >
    <div
      class="m_e17b862f mantine-NavLink-children"
    />
  </div>
</nav>
`;
